<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="enabled">
   <bool>true</bool>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>918</width>
    <height>748</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>918</width>
    <height>748</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>918</width>
    <height>748</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Graphics</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QCustomPlot" name="customPlot" native="true">
    <property name="geometry">
     <rect>
      <x>-1</x>
      <y>19</y>
      <width>511</width>
      <height>671</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_Ok">
    <property name="geometry">
     <rect>
      <x>740</x>
      <y>330</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>OK</string>
    </property>
   </widget>
   <widget class="QTextBrowser" name="textBrowser">
    <property name="geometry">
     <rect>
      <x>520</x>
      <y>390</y>
      <width>191</width>
      <height>111</height>
     </rect>
    </property>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>520</x>
      <y>31</y>
      <width>389</width>
      <height>276</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout">
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_2">
       <item>
        <widget class="QLabel" name="label">
         <property name="text">
          <string>Method</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QComboBox" name="comboBox_metod">
         <item>
          <property name="text">
           <string>Coordinate Descent</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Stohastic Search</string>
          </property>
         </item>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_3">
       <item>
        <widget class="QLabel" name="label_5">
         <property name="text">
          <string>Function</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QComboBox" name="comboBox_function">
         <item>
          <property name="text">
           <string>(x - 2)^4  + (x - 2*y)^2</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>0.1*cos(10*(x^2 + y^2))</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>x^2 + y^2</string>
          </property>
         </item>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_4">
       <item>
        <widget class="QLabel" name="label_6">
         <property name="text">
          <string>Stop criterion for CD</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QComboBox" name="comboBox_Stop_crit_CD">
         <item>
          <property name="text">
           <string>||x_{k} - x_{k-1}|| &lt; eps</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>|| gr(F_k)|| &lt; eps</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>||(F_{k} - F_{k-1})/F_{k}|| &lt; eps</string>
          </property>
         </item>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_5">
       <item>
        <widget class="QLabel" name="label_7">
         <property name="text">
          <string>Stop criterion for SS</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QComboBox" name="comboBox_Stop_crit_SS">
         <item>
          <property name="text">
           <string>|(F_{k+j} - F_{k}| &lt; eps</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Limit N on the numb of iter after the last improvement</string>
          </property>
         </item>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_9">
       <item>
        <widget class="QLabel" name="label_8">
         <property name="text">
          <string>Area</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QComboBox" name="comboBox_Area">
         <item>
          <property name="text">
           <string>[0 ; 3]^2</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>[0 ; 3]*[0; 6]</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>[-5 ; 5]^2</string>
          </property>
         </item>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout">
       <item>
        <widget class="QLabel" name="Print_N">
         <property name="text">
          <string>Print N</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSpinBox" name="Print_N_2">
         <property name="minimum">
          <number>1</number>
         </property>
         <property name="maximum">
          <number>1000</number>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_6">
       <item>
        <widget class="QLabel" name="label_2">
         <property name="text">
          <string>Local probability for SS metod</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QDoubleSpinBox" name="prob">
         <property name="decimals">
          <number>6</number>
         </property>
         <property name="minimum">
          <double>0.100000000000000</double>
         </property>
         <property name="maximum">
          <double>1.000000000000000</double>
         </property>
         <property name="singleStep">
          <double>0.000001000000000</double>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_7">
       <item>
        <widget class="QLabel" name="label_3">
         <property name="text">
          <string>Enter eps for Stop criterion</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QDoubleSpinBox" name="eps">
         <property name="decimals">
          <number>7</number>
         </property>
         <property name="minimum">
          <double>0.000001000000000</double>
         </property>
         <property name="maximum">
          <double>1.000000000000000</double>
         </property>
         <property name="singleStep">
          <double>0.000001000000000</double>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_8">
       <item>
        <widget class="QLabel" name="label_4">
         <property name="text">
          <string>Enter maximum number of iterations</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSpinBox" name="max_iter">
         <property name="minimum">
          <number>10</number>
         </property>
         <property name="maximum">
          <number>10000</number>
         </property>
         <property name="singleStep">
          <number>10</number>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_10">
       <item>
        <widget class="QLabel" name="label_9">
         <property name="text">
          <string>X_0</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QDoubleSpinBox" name="x_0">
         <property name="decimals">
          <number>6</number>
         </property>
         <property name="minimum">
          <double>-99.000000000000000</double>
         </property>
         <property name="singleStep">
          <double>0.001000000000000</double>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_10">
         <property name="text">
          <string>Y_0</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QDoubleSpinBox" name="y_0">
         <property name="decimals">
          <number>6</number>
         </property>
         <property name="minimum">
          <double>-99.000000000000000</double>
         </property>
         <property name="singleStep">
          <double>0.001000000000000</double>
         </property>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>918</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QCustomPlot</class>
   <extends>QWidget</extends>
   <header>qcustomplot.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
